src/app/packages/open-layers-map/image-processing/open-layers-image-processing.ts

```diff
@@ -1,43 +1,77 @@
 import Raster from 'ol/source/raster';
 
+export type supportedOperations = 'Histogram_Auto' | 'Histogram_manual' | 'Contrast' | 'Brightness' | 'Sharpness';
+
+export const IMG_PROCESS_ORDER = ['Histogram_Auto', 'Histogram_manual', 'Contrast', 'Brightness', 'Sharpness', ];
+
+export const IMG_PROCESS_DEFAULTS = {
+	SHARPNESS: 50,
+	HISTOGRAM_LOW: 0,
+	HISTOGRAM_HIGH: 100,
+	CONTRAST: 50,
+	BRIGHTNESS: 50
+};
 export type pixelOperation = (pixels: ImageData[], data: Object) => (ImageData);
 
 export interface IRasterOperation {
 	name: string,
 	operation: pixelOperation,
+	value?: any,
+	defaultValue: any,
 	lib: {}
 }
 
-export type supportedOperations = 'Histogram' | 'Sharpness';
 
 export const operations = [];
 
 export class OpenLayersImageProcessing {
 
 	_rasterToOperations: Map<Raster, IRasterOperation[]>;
 	_operations: Map<string, IRasterOperation>;
+	_operationsAsStrings: Map<string, string>;
+	_libsAsStrings: Map<string, string>;
 
 	constructor() {
 		this._rasterToOperations = new Map<Raster, IRasterOperation[]>();
 		this._operations = new Map<string, IRasterOperation>();
+		this._operationsAsStrings = new Map<string, string>();
+		this._libsAsStrings = new Map<string, string>();
 
 		this.initializeOperations();
 	}
 
 	initializeOperations() {
-		this.initializeHistogramEqualization();
+		this.initializeAutoHistogramEqualization();
 		this.initializeSharpness();
+		this.initializeContrast();
+		this.initializeBrightness();
+	}
+
+	addOperation(operation) {
+		this._operations.set(operation.name, operation);
+		this._operationsAsStrings.set(operation.name, operation.operation.toString());
+		for (let property in operation.lib) {
```

Comment:
Object.keys().forEach to avoid hasOwnProperty

Reply:
done